### PR TITLE
TOML formatting using pyproject-fmt

### LINKED ISSUE #212: TOML formatting
This question still doesn’t seem exhaustively answered: https://github.com/toml-lang/toml/issues/532

The most promising option right now seems to be: https://github.com/tamasfe/taplo

### PR DESCRIPTION
**Pull request by flying-sheep**
Fixes  #212

An alternative to #225, see #212 for a comparison

### CODE REVIEW

File: biome.jsonc
Comment (flying-sheep) [inline biome.jsonc:7]: we could also leave it out, then all JSON files get formatted with tabs

Comment (grst) [reply biome.jsonc:7]: Let's keep it, too bad biome doesn't support editorconfig.

File: {{cookiecutter.project_name}}/pyproject.toml
Comment (grst) [inline {{cookiecutter.project_name}}/pyproject.toml:135]: Are all these rearrangements necessary? This will create one huge merge conflict when making the template-update PRs

**Comment by grst** [comment]
I like it in principle! Shall we wait a bit longer until the biome-pre-commit and pyproject-fmt are maturing a bit more?
How likely is it that pyproject-fmt will become part of ruff?

File: {{cookiecutter.project_name}}/pyproject.toml
Comment (flying-sheep) [reply {{cookiecutter.project_name}}/pyproject.toml:135]: That’s the thing: pyproject-fmt dictates the order of these blocks.

If it was possible to run the hooks before creating the diff, maybe we could avoid conflicts

**Comment by flying-sheep** [comment]
> Shall we wait a bit longer until the biome-pre-commit and pyproject-fmt are maturing a bit more?

I think pyproject-fmt is mature enough, for biome-pre-commit we should maybe wait until someone replies to my comment in https://github.com/biomejs/pre-commit/issues/5

> How likely is it that pyproject-fmt will become part of ruff?

it’s a completely separate rust project, and targets a different type of file, so maybe not that likely?

but then again the ruff people have adopted `uv` and `rye` so who knows!

File: biome.jsonc
Comment (flying-sheep) [reply biome.jsonc:7]: It’s coming, actually! https://github.com/biomejs/biome/issues/1724

**Comment by flying-sheep** [comment]
I’ll split this up into two PRs

**Review by grst — APPROVED** [review]
I think I'd prefer this one over taplo - In addition to the reasons you mentioned it
 * doesn't require another `.<formatter>` config file
 * seems to do more stuff that's specific to pyproject.toml, but useful (such as sorting dependencies)

**Comment by flying-sheep** [comment]
OK! Do you think we should leave it unconfigured (2 spaces indent) or add

```toml
[tool.pyproject-fmt]
indent = 4
```

**Comment by flying-sheep** [comment]
Welp, we should decide on indents before next release, but I’m going to merge this

**Pull request merged**
